### ISSUE TITLE
Shifting to 1.12

### ISSUE THREAD

**Issue by drrugg** [scalda/FoolCraft#889]
## SUGGESTION/HELP
- [X ] I would like to propose an additional mod, config change, custom recipe, etc.
- [ ] I am requesting help with installing or running the pack

## Issue Description
What are the core mods that prevent FC3 being on 1.12?

Several mods this year have discontinues 1.10 support. BOP's nether and Astral are particularlly hampered. These are the primary ones that come to mind;
* Applied Energistics
* Astral Sorcery
* Biomes o Plenty
* Botania
* JEI
* Tinkers Contruct
* Weystones
* Woot

Some obvious core "fool" mods are supported as 1.12. And obviously some mods like EnderIO are only 90% converted over. Previous bug reports just mention waiting on 'some' mods.

## Reproduce Steps

Download Foolcraft, and play along to several seasons with the Hermits. Love foolcraft. Briefly have people make you serve ATM3 instead. Return to Foolcraft, as the great game that the amazing Scalda and the other devs and admins have lovingly hand crafted. Try to figure out what the best way to support the dev and continuation of this pack.

## Additional Information

Long time lurker, and small kiwi foolcraft server runner, and IRL 'cloud' admin and PS scripter. Recent iskall discord lurker.

If a version tracking list doesn't already exist, I certainly would love to help out with putting one together. Certainly would love to throw my hand in and help out with comms, or anything, but no idea where to start? I certainly have a growing interest in minecraft server maintenance, and a working knowledge of change/release management ;)

**Comment by svennieke** [comment]
i think one of the biggest issues of just updating from 1.10 to 1.12 is that mods have changed, their ID's have changed, things  work differently, etc meaning a lot of things in foolcraft need to be redone. i would personally suggest moving to 1.12 if there ever comes a foolcraft 3, this would also prevent the issue of the arena corrupting and such, in most cases it is not as easy as just updating the mods to a newer mc version

**Comment by drrugg** [comment]
Certainly thinking FC3 rather than an upgrade path - sorry it was unclear.

**Comment by pinkphoenix1990** [comment]
Even outside of mod developers switching to actively maintaining 1.12, the base Vanilla version of 1.12 adds a lot of great quality of life features (i.e. crafting book, items from crafting tables going back into inventory instead of being dumped, etc.). Besides, I'm thinking that the other mods will finally be fully updated to 1.12 by the time a FC3 could be considered!

**Comment by Mrbysco** [comment]
There's one issue with 1.12 mod packs that I noticed. In 1.12 I've experienced issues with crafting in heavily modded packs, Issues like shift clicking lagging the server because of the amount of recipes added by mods.

**Comment by jmac460** [comment]
I've noticed a lot of mods that were made for 1.10.2, have lots of bugs that were said to be fixed in the 1.12 version... except im playing FC2 so i cant actually get that version. Hopefully they make the change to 1.12.... i really need it for fps sake and other bugs sake

**Comment by Mrbysco** [comment]
Another point I've noticed in 1.12 is that it's much heavier on computers. 
What could run with 3gb in 1.10 can take up to 4+ gigs in 1.12.

So it's the question if people who can barely play the 1.10 can even play on 1.12

**Comment by Mrbysco** [comment]
I've made a spreadsheet with the list of mods that are in Foolcraft and if they have a 1.12 version of equivalent.

https://docs.google.com/spreadsheets/d/1k8IaQuVn1o6QK3sDO0UZFUiUrAOJGODE_biqJkxid_I/edit?usp=sharing

**Comment by drrugg** [comment]
Thanks @Mrbysco - that's awesome! I hadn't realized how close it might be, esp. with your annotations. Could I suggest 'dark green' for 'obsolete' just to visually show how close it might be.

**Comment by Mrbysco** [comment]
The spreadsheet was made because I was bored and had time to kill. and also because I wanted to know which mod were yet to update

**Comment by drrugg** [comment]
~80%, 4% partially available, several being written.  Cool "temporary sort" - select C1-C146, hit 'filter', only show 'No' - shows the 27 mods missing 1.12.

There are the obvious two "Hermit" mods, and possibly migration to 'Industrial Foregoing'.

Is this something of interest to the mods and admins? I don't know even if a Foolcraft3 v0.1a might be viable?

**Comment by AutisticMC** [comment]
I love the idea

**Comment by AutisticMC** [comment]
And plz add JurassicCraft and Fossil and Archeology Because it's These are cool mods

**Comment by notmark** [comment]
Personally I would wait until 1.13 is out since it is a major technical update to the game itself. Stick with 1.10.2 for the time being.

**Comment by Mrbysco** [comment]
The moment 1.13 comes out it will take MONTHS for updates and even longer to become stable.
And since most devs stopped supporting 1.10 I'd go to 1,12 just for having support from the devs.

**Comment by foxanthony1** [comment]
The Fools ran into many bugs when they started playing FC1 & 2. It could be helpful to release FC3/1.12  as an early beta for public testing; so they could play a more stable pack when they do start playing it.

**Comment by drrugg** [comment]
tldr; Success. More than I imagined.


So I finished upgrading our kiwi server foolcraft 2.6 through dev into prod, and finished putting together a custom profile in Curse/Twitch just picking up mods that are on 1.12. Singleplayer runs as badly as FC2.6 on my client. Spun up a new server instance in AMP, using forge+library + server files from ATM3 server, mods from the custom profile, and configs from the main pack.

Works like a dream. Seriously good, after I dropped ModernWarefare. I hit creative, and flew a tour around in a Vanilla and BIOMESOP worlds, tpbiome to my favorites, including the five Nether biomes. Rushed AstralSourcery to see the fifth chapter. Saw RoguelikeDungeon's the new wizard tower and forest building. Played around with giant chance cubes, dodecahedron, the far improved warp stones interface. On the technical side, hardly any lag - performed much the same as my 1.10 client and server would usually do. Only after hitting the nether did my client glug up and need me to restart Java - no server issues at all. A few "Flying too fast" warning, and disableElytraMovementCheck rule didn't help.

Then I took a copy-pasta 1.10 world into the 1.12 world folder. And ... it worked. Well, i got plenty of tick killing "skipping entity", and the server crashed frequently BUT only because of trying to assign jobs to zombie villagers every 10s or so. But I did log into my usual base, saw my IE2 system still in place, and between server reboots flew and generated a few new chunks with no hard biome change and new 1.12 mod features being generated. That blows my mind.

Its obviously missing the tweaks, optimizations, and combinations the masters and crafters of the pack do. It is *really* tempting to put together an exported Curse profile for my players to run. Its that good.

I'll echo @foxanthony1 's suggestion - what would it take to create a beta curse profile? Or for putting me/our hand in? It is seriously that good, fun, and up to date with all the latest changes.

**Comment by Mrbysco** [comment]
Recently Avaritia and Hatchery updated. So I updated the spreadsheet.

Also I've been spending today copying the configurations that were made in foolcraft to my 1.12 clone.
(got most of the scripts working)

**Comment by Mrbysco** [comment]
I've also managed to get most of the quests to work.
Obviously there are missing item rewards due to missing mods. But most are functional.

**Comment by drrugg** [comment]
Awesomesause @Mrbysco! Is there anything I can assist with tweaking the pack? I was kind of hoping I could offer @scalda etc help but they are likely busy with other projects.

Could I suggest for your spreadsheet a column confirming mods that are 1.10 unsupported?

Couple of updates;
- EnderIO - Discord pin post (Dec13) focused on EnderCore only. "~95% for base, ~85% for machines and ~12% for conduits" (conduits being least to do with EnderCore) but quite active/excited conversations.
- MiniMoos - DarkVicious "The 1.12 Version is being worked on and should become available in January"
- Railcraft  1.11/1.12 migration list - https://github.com/Railcraft/Railcraft/issues/1232
- SimplyConveyors is 1.12 only now and requires CompatLayer for 1.11 below. 
  - Does that mean CompatLayer could help 1.10 mods in the meantime?? Or is that the wrong direction.
- SubLib comments points to a 1.12.0 version, Subaraki has been off the scene a while. I also can't see anything in Dependents (maybe it was a removed mod?) - probably can write it off for now
- Winter Grave (zarius repo?) in a Oct youtube was saying he's overloaded, and his name is against Iskallium Reactors, Hermitron, WeaponCase, but has ported his Subterranean Descent to 1.12.


Hearing you have a mostly working clone is great to hear. I hope awesome things happen now its `18! Happy new year!

**Comment by Mrbysco** [comment]
![afbeelding](https://user-images.githubusercontent.com/9308260/34525363-f9cf5446-f09e-11e7-8dcd-a92835088a4c.png) Compatlayer is to make 1.10 and 1.11 versions compatable. Not to make 1.12 work in lower.

**Comment by drrugg** [comment]
Love your work. Would I be able to acquire a copy of the file tweaks that you have made?

I made the discovery that Sponge hasn't been supporting 1.10 Forge for the last two versions either. Forge is in support phase only (reminds me of Microsoft Lifecycles).

**Comment by Mrbysco** [comment]
Angel ring to bauble has gotten an update. And the weirding gadget got a temporary workaround to fix not being able to use forge 2576+.

**Comment by generrosity** [comment]
Vic's Modern Warfare Mod  isn't on your list, and is 1.12 updated (in the 1.11 jar apparently).

**Comment by generrosity** [comment]
Iskallium Reactores now has 1.11 and 1.12 versions. Thaumcraft to go on the list - once he has 1.10 out of beta he starts on 1.12

I'm still keen if you could share those 1.12 tweaks youve discovered - it would be interesting if Scalda could support two pack versions at once and they do migrate as easy as I have tried so far.

**Comment by scalda** [comment]
I will start looking at 1.12 next month perhaps, i need to have a chat with Iskall regarding this

**Comment by generrosity** [comment]
Best post to wake up to on a lazy Saturday morning. Thank you for considering!

**Comment by scalda** [comment]
lol no problem

**Comment by generrosity** [comment]
The great head ring leader had whispered about a FoolCraft3... I can hope for 1.12, and wish for rumors? 😜 Great to see him fired up and his patreons involved, looking forward to the latest server files/log with ProjectE.

**Comment by generrosity** [comment]
I see Mischief of Mice has done a 1.12 Ender IO review 😯 And I see on they are quickly moving to support 1.13. 

Perhaps Foolcraft 3 is 1.13? Eagerly waiting on FoolCraft2.13 server files in the meantime 😜
